src/test/groovy/org/prebid/server/functional/tests/TargetingSpec.groovy

```diff
@@ -418,19 +420,20 @@ class TargetingSpec extends BaseSpec {
         assert response.targeting.isEmpty()
     }
 
-    def "PBS amp should make right calculation for hb_pb targeting"() {
+    def "PBS amp should use ranges.max value for hb_pb targeting when bid.price is greater than ranges.max"() {
         given: "Default amp request"
         def ampRequest = AmpRequest.defaultAmpRequest
 
         and: "Default bid request with targeting and stored bid response"
         def storedBidResponseId = PBSUtils.randomString
+        def max = PBSUtils.randomDecimal
+        def precision = 2
         def ampStoredRequest = BidRequest.defaultBidRequest.tap {
             imp[0].ext.prebid.storedBidResponse = [new StoredBidResponse(id: storedBidResponseId, bidder: GENERIC)]
             ext.prebid.targeting = Targeting.createWithAllValuesSetTo(true).tap {
                 priceGranularity = new PriceGranularity().tap {
-                    precision = 2
-                    ranges = [new Range(max: 1.50,increment: 1.0),
-                              new Range(max: 2.50,increment: 1.2)]}
+                    it.precision = precision
+                    ranges = [new Range(max: max, increment: PBSUtils.randomDecimal)]}
```

Comment:
Are you sure that it will not be flickering when random increments are bigger than random max?

Reply:
Yes, it gonna work without issue